crates/interpreter/src/valid.rs

```diff
@@ -412,31 +423,42 @@ struct Expr<'a, 'm> {
     side_table: SideTable,
 }
 
+#[allow(dead_code)]
+#[derive(Default)]
+struct ParserRange {
```

Comment:
Unless we want to implement methods on this, we can just use `Range<usize>` whenever we need a `ParserRange`. This is only used locally and the variable name should always clarify the meaning.